### PR TITLE
Added class offering link click logging [#174653677]

### PR DESCRIPTION
**Pull request by dougmartin**
This adds a generic logger client to the portal with the offering details component as the first user of the client.

The following events are logged:

- clickedPreviewLink
- clickedTeacherEditionLink
- clickedReportLink
- clickedExternalReport

in each case the event_value is the url of the link.

### CODE REVIEW

**Comment by dougmartin** [comment]
@scytacki A couple of notes:

1. This defines a new application name ('rigse-log') for the portal log messages
2. I wasn't sure if we wanted to use snake_case or camelCase for the event names so I went with cameCase.

File: react-components/src/library/components/assigments/offering-details.js [hunk 1]
```diff
@@ -1,32 +1,64 @@
...
           reportUrl &&
-          <a href={reportUrl} target='_blank' className={commonCss.smallButton} title='Report'>Report</a>
+          <a href={reportUrl} target='_blank' className={commonCss.smallButton} title='Report' onClick={() => logEvent(reportLogData)}>Report</a>
         }
         {
           externalReports && externalReports.map((externalReport, index) => {
+            const externalReportLogData = {
+              event: 'clickedExternalReport',
+              event_value: externalReport.url
+            }
```
Comment (scytacki) [inline react-components/src/library/components/assigments/offering-details.js:55]: This seems to be missing the activity, and parameters that other events have.

The activity of these report launches is going to have to change at some point, but that can be done in a later PR and using `activity: [id]` seems good enough for now.

**Review by scytacki — CHANGES_REQUESTED** [review]
looks good, just one oversight in external report handling I think.

**Comment by dougmartin** [comment]
@scytacki I added a follow on commit to add the activity and parameters to the other event logged: https://github.com/concord-consortium/rigse/pull/925/commits/9b28ece75471d5708f68d4e0cddbf630adaf3486

**Review by scytacki — APPROVED** [review]
looks good to me

**Pull request merged**